File: src/frontend/src/app/components/financeiro/condicao-pagamento/model/condicao-pagamento-grid.dto.ts
import { Exclude, Expose } from 'class-transformer';

@Exclude()
export class CondicaoPagamentoGridDTO {
  @Expose()
  id?: string;

  @Expose()
  condicao: string;

  @Expose()
  ativo: boolean;

  @Expose()
  quantidadeParcelas?: number;

  @Expose()
  deleted?: boolean;

  constructor(
    condicao: string,
    ativo: boolean,
    quantidadeParcelas?: number,
    id?: string,
    deleted?: boolean
  ) {
    this.condicao = condicao;
    this.ativo = ativo;
    this.quantidadeParcelas = quantidadeParcelas;
    this.id = id;
    this.deleted = deleted;
  }
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/financeiro/condicaopagamento/CondicaoPagamentoGridDTO.java
package br.com.grupopipa.gestaointegrada.financeiro.condicaopagamento;

import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class CondicaoPagamentoGridDTO implements GridDTO {
    private UUID id;
    private String condicao;
    private Boolean ativo;
    private Integer quantidadeParcelas;
    private Boolean deleted;
}
